src/infra/flow-system-inmem/tests/tests/utils/task_driver.rs

```diff
@@ -86,17 +87,19 @@ impl TaskDriver {
             .is_some()
     }
 
-    async fn ensure_task_matches_dataset(&self) {
+    async fn ensure_task_matches_logical_plan(&self) {
         let task = Task::load(self.args.task_id, self.task_event_store.as_ref())
             .await
             .expect("Task does not exist yet");
 
+        assert_eq!(self.args.expected_logical_plan, task.logical_plan);
         match &task.logical_plan {
             LogicalPlan::UpdateDataset(ud) => {
                 assert!(self.args.dataset_id.is_some());
                 assert_eq!(&ud.dataset_id, self.args.dataset_id.as_ref().unwrap());
             }
             LogicalPlan::Probe(_) => assert!(self.args.dataset_id.is_none()),
+            LogicalPlan::HardCompactDataset(_) => (),
```

Comment:
Don't you have a dataset id  there to assert?

Reply:
Also, where do you check the configuration options now?

Reply:
I'm asserting full logical plan above. Not sure we need to assert additionally `datase_id` here